models/login_source.go

```diff
@@ -642,6 +642,12 @@ func UserSignIn(username, password string) (*User, error) {
 	}
 
 	if hasUser {
+		if !user.IsActive {
```

Comment:
This should be checked only after username&password is validated otherwise usernames can be leaked/guessed from private servers